### DISCUSSION TITLE
Cannot instantiate `BaseSQLToGCSOperator`

### DISCUSSION THREAD

**Discussion by lafet1** [apache/airflow#25477] [Q&A] [answered]
Hello everybody,

this is my first issue/question. If I'm asking something already answered/solved, sorry and could you, please, direct me to where I can find the answer/solution?

I'm using Airflow 2.2.1 and I want to use the operator `BaseSQLToGCSOperator` to export a subset (2 out of 4 columns) from BigQuery to GCS. The issue is [this line of code](https://github.com/apache/airflow/blob/10b198ac5b8451bf753f4a53ef59907d47005d96/airflow/providers/google/cloud/transfers/sql_to_gcs.py#L310),  which I do not fully understand. This makes the class abstract and blocks me from instantiating it. My question is why? Is this intentional (most likely) or a bug? And if it is intentional why is it easily accessible such as this and even in the docs without the mention of the fact that it is abstract?
* I'm assuming it is intentional and it has to do with connecting and correctly loading from a specific SQL database. But again, not hundred percent certain here.

I'm not using `BigQueryToGCSOperator` as this operator cannot subset the table. I would have to create an extra table which I of course can do but would prefer not to.

Is there a way to export a table from BigQuery to GCS **while** applying only a simple column subset or not?

Thanks in advance and if some information is missing, please, let me know. I'll add it.

**Comment by potiuk** [comment, marked as answer]
Base* is usually a prefix for abstract operators that other operators make "concrete". Why do you have problem with using any other derived operators from it ? What stops youfrom doing it? I do not understand what you mean by 'operator cannot subset the table'?

**Reply by lafet1** [reply]
Thanks for the quick answer!

I want to export data from BigQuery and there is no SQL operator for that. Or at least I can't find it, the derived operators are for MSSQL, Oracle, Trino, Presto and Postgres but not for BigQuery. [This operator](https://airflow.apache.org/docs/apache-airflow-providers-google/stable/_api/airflow/providers/google/cloud/transfers/bigquery_to_gcs/index.html) isn't derived from this SQL base operator.

> I do not understand what you mean by 'operator cannot subset the table'?

I meant that the only BigQuery to GCS operator is a [transfer operator](https://airflow.apache.org/docs/apache-airflow-providers-google/stable/_api/airflow/providers/google/cloud/transfers/bigquery_to_gcs/index.html) and not an SQL one. It doesn't have a parameter for `sql` code or to subset any columns. Because that's what I was hoping to do, send only two columns from BQ to GCS.

P.S. On the plus side I think I finally understand my confusion in the beginning. Thanks for helping me orient in this!

**Reply by potiuk** [reply]
Well. You can always write the operator yourself basing on the code of other similar operators and even contribute it back. Airlfow is created by > 2100 contributors - most like you who wanted to have a feautre and added it. Developing Custom operators is a 1st-class citizen in Airflow and it is not too complex, not more complex than creating DAGs - and you have souce code for the other operators, so you can easily use them as a base.
